third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/BarrierOpToLLVM.cpp

```diff
@@ -152,25 +157,56 @@ struct WaitBarrierOpConversion
         typeConverter->convertType(op.getAlloc().getType().getElementType()),
         rewriter);
     auto loc = op.getLoc();
-    const std::string ptxNoPred =
-        "{                                                           \n\t"
-        ".reg .pred P1;                                              \n\t"
-        "waitLoop:                                                   \n\t"
-        "mbarrier.try_wait.parity.shared.b64 P1, [$0], $1;           \n\t"
-        "@!P1 bra.uni waitLoop;                                      \n\t"
-        "}                                                           \n\t";
-    const std::string ptxPred =
-        "{                                                           \n\t"
-        "@!$2 bra.uni skipWait;                                      \n\t"
-        ".reg .pred P1;                                              \n\t"
-        "waitLoop:                                                   \n\t"
-        "mbarrier.try_wait.parity.shared.b64 P1, [$0], $1;           \n\t"
-        "@!P1 bra.uni waitLoop;                                      \n\t"
-        "skipWait:                                                   \n\t"
-        "}                                                           \n\t";
-    ::mlir::triton::PTXBuilder ptxBuilder;
     bool predicated = adaptor.getPred() != nullptr;
-    std::string ptx = predicated ? ptxPred : ptxNoPred;
+    std::string ptx;
+    if (targetInfo->getComputeCapability() < 90) {
+      if (!predicated) {
+        ptx = R"(
+{
+	.reg .pred complete;
+	waitLoop:
+	mbarrier.test_wait.parity.shared.b64 complete, [$0], $1;
+	@!complete nanosleep.u32 20;
```

Comment:
Huh, how come this wasn't needed before for Ampere?

Reply:
I think we just didn't use mbarrier on ampere before and always used the `commit_group`/`wait_group` based synchronization.

Reply:
Ah of course. Makes sense.